src/components/molecules/OfferDetails/OfferDetails.tsx

```diff
@@ -0,0 +1,281 @@
+                      cursor: 'pointer',
+                    }),
+                  }}
+                  onClick={() =>
+                    isVideoCall && typeof onPopupClick === 'function' ? onPopupClick(true) : null
+                  }
+                >
+                  <Text mr={isVideoCall ? 2 : 0}>Rekrutacja online</Text>
+                  {isVideoCall && (
+                    <Icon
+                      name="Info"
+                      sx={{
+                        width: '16px',
+                        height: '16px',
+                        flexShrink: 0,
+                        bg: 'secondary',
+                        borderRadius: '50%',
+                        color: 'white',
+                      }}
+                    />
+                  )}
+                </Tag>
+              </Tooltip>
+            )}
+            {!offerExpired && subtitle}
+          </Flex>
+        </Flex>
+        <Box
+          sx={{
+            position: 'relative',
+            width: 'calc(100% + 32px)',
+            height: 0,
+            overflow: 'hidden',
+            pt: '53%',
+            ml: -3,
+          }}
+        >
+          {typeof offerImage === 'string' ? (
+            <>
+              <Image
+                sx={{
+                  position: 'absolute',
+                  top: 0,
+                  left: 0,
+                  width: '100%',
+                  height: '100%',
+                  objectFit: 'cover',
+                  ...(offerExpired && {
+                    opacity: 0.2,
+                    filter: 'grayscale(1)',
+                  }),
+                }}
+                src={offerImage}
+              />
+              {company.isGoodCompany && (
+                <Icon
+                  name="GoodEmployer"
+                  sx={{
+                    position: 'absolute',
+                    right: ['16px', '24px'],
+                    bottom: ['16px', '24px'],
+                    width: '145px',
+                    height: 'auto',
+                    borderRadius: 1,
+                    boxShadow: 1,
+                    filter: offerExpired ? 'grayscale(1)' : 'initial',
+                  }}
+                />
+              )}
+            </>
+          ) : (
+            offerImage
+          )}
+        </Box>
+        <Box width="100%">
+          <Flex mt="20px" color={getBadgeColor('FFDFDF')}>
+            <Icon name="Location" />
+            <Flex flexDirection="column" ml={2} sx={{ overflow: 'hidden', color: 1 }}>
+              <Text sx={{ fontScale: 4, fontWeight: 'bold' }}>Adres</Text>
+              <Text>{address}</Text>
+              <Box>
+                {!!calculated_distance && (
+                  <Text sx={{ fontWeight: 'bold', fontScale: 2 }}>
+                    {formatDistance(calculated_distance)} km od Ciebie{` `}
+                  </Text>
+                )}
+                <Text
+                  as={!!onMapClick ? 'button' : 'span'}
```

Comment:
`!!` jest niepotrzebne przy ternary

Reply:
jeśli nie ma onMapClick to można zostawić to jako button, tylko po prostu dać może pointerEvents: 'none' albo żeby funkcja na onClick po prostu nic nie robiła?

Reply:
czemu button ma być buttonem skoro nie spełnia takiej funkcji?

Reply:
bo strasznie tu dużo warunków, są w: 
1. as
2. onClick
3. sx

Może się mylę, ale wystarczyłoby w sx dać pointerEvents: 'none' gdy nie ma onMapClick'a albo default props dla onMapClick, wtedy nie trzeba żadnego warunku, przycisk będzie ale nie bedzie nic robił. W widoku podglądu oferty nie ma to znaczenia.

Reply:
nie chciałam, żeby był buttonem, skoro nie ma onClick, ale Twoje argumenty mnie przekonały :D